packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2418,11 +2420,34 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     updateKeepAlive();
   }
 
+  String _cachedText = '';
+  Rect? _cachedFirstRect;
+
+  void _updateSelectionRects() {
+    if (defaultTargetPlatform != TargetPlatform.iOS)
+      return;
+    final TextSpan textSpan = buildTextSpan();
+    final StringBuffer text = StringBuffer();
+    textSpan.computeToPlainText(text);
+    final Rect firstRect = renderEditable.getBoxesForSelection(const TextSelection(baseOffset: 0, extentOffset: 1)).first;
+    final ScrollDirection scrollDirection = _scrollController?.position.userScrollDirection ?? ScrollDirection.idle;
+    if (scrollDirection == ScrollDirection.idle && (text.toString() != _cachedText ||
```

Comment:
`TextSpan` has `computeToPlainText`, which accepts a `StringBuffer`, so `.toString()` here is actually called on `StringBuffer`

For `ScrollDirection`, the idea is to wait to send updated selection rects until scrolling stops – that way large amounts of text don't cause scrolling to be choppy.

That's a good point on the cache invalidation...the size is easy enough to check, I suppose, are there any other invalidation conditions you can think of?

Reply:
`toString` produces debug messages for the object so in the future more stuff might be added to it. I think `computeToPlainText` might be the more robust choice. 

> are there any other invalidation conditions you can think of

If the paragraph starts with a zero-width glyph (for example, line feed), it's not going to represent the layout of other glyphs well.

Reply:
Gotcha, well, it is using `computeToPlainText` so I think we're OK on that. For the zero width glyph...that might be OK, I think I was just using the firstRect as a stand in for the offset of the text, since if it gets scrolled, the rects will need to be recalculated. Maybe it's OK without that? I'll have to do a check with some longer text and see.